### ISSUE TITLE
Adding backends with specific rules make first backend unreachable

### ISSUE THREAD

**Issue by victorklos** [containous/traefik#1070]
### What version of Traefik are you using (`traefik version`)?

v1.1.2 / camembert

### What is your environment & configuration (arguments, toml...)?

Using docker 1.12.2:

```
    docker run --name=traefik --restart=always \
        -p 80:80 -p 127.0.0.1:8080:8080 \
        -v /var/run/docker.sock:/var/run/docker.sock \
        -v /dev/null:/traefik.toml                                            
```

### What did you do?

Started two `nginx` docker containers with the following labels:
```yml
    traefik.port=80
    traefik.backend=lunawebsite
    traefik.frontend.rule=Host:luna.liefdevolafscheid.nl

    traefik.port=80
    traefik.backend=luna-redirect
    traefik.frontend.rule=Host:luna-liefdevolafscheid.nl
```

(Mind the dot versus dash in the hostname.)

### What did you expect to see?

Two backends plus two frontends. The first would serve the website http://luna.liefdevolafscheid.nl/ (the preferred URL) and the second would serve http://luna-liefdevolafscheid.nl/ but redirect most of the content to the preferred website.

### What did you see instead?

A single frontend plus two backends. The preferred website (the first nginx container) is no longer reachable as it doesn't have a corresponding frontend.

### What do you think has happened?

It seems that both `traefik.frontend.rule=Host:luna.liefdevolafscheid.nl` and `traefik.frontend.rule=Host:luna-liefdevolafscheid.nl` result in a Route with the name (`route-frontend-Host-luna-liefdevolafscheid-nl`) so that what should be adding a Route becomes overwriting.

**Comment by ldez** [comment]
Do you still have the same problem with the latest version? [releases](https://github.com/containous/traefik/releases)

**Comment by stevejr** [comment]
Is there an update on this issue?

It seems that if you do not use a unique value in the front end rule i.e different Hosts then the rule is just created with the same name.

**Comment by victorklos** [comment]
Tested this again on version `1.4.0 roquefort` and I can confirm that this is still an issue.

As the route name is determined from the rule a workaround is to add a bogus hostname as unique token before the real host, e.g. as follows:

```
    traefik.port=80
    traefik.backend=lunawebsite
    traefik.frontend.rule=Host:abc,luna.liefdevolafscheid.nl

    traefik.port=80
    traefik.backend=luna-redirect
    traefik.frontend.rule=Host:xyz,luna-liefdevolafscheid.nl
```

Not very elegant but it works for me.

**Comment by traefiker** [comment]
Closed by #2338.
